### ISSUE TITLE
unchecked warnings in REPL of recent 2.13.x build

### ISSUE THREAD

**Issue by Jasper-M** [scala/bug#11902]
```scala
Welcome to Scala 2.13.2-bin-f86e6b (Java HotSpot(TM) 64-Bit Server VM, Java 1.8._152).
Type in expressions for evaluation. Or try :help.

scala> :pa
// Entering paste mode (ctrl-D to finish)

sealed abstract class Value
object Value {
    final case class Num(value: Double) extends Value
    final case class Str(value: String) extends Value
    final case class Bool(value: Boolean) extends Value
}

// Exiting paste mode, now interpreting.

warning: 3 unchecked warnings; for details, enable `:setting -unchecked' or `:replay -unchecked'
class Value
object Value

scala> :warn
           final case class Num(value: Double) extends Value
                            ^
On line 3: warning: The outer reference in this type test cannot be checked at run time.
           final case class Str(value: String) extends Value
                            ^
On line 4: warning: The outer reference in this type test cannot be checked at run time.
           final case class Bool(value: Boolean) extends Value
                            ^
On line 5: warning: The outer reference in this type test cannot be checked at run time.
```

Those warnings weren't there in 2.13.1.

**Comment by dwijnand** [comment]
Same as #11534? (via @smarter)

**Comment by som-snytt** [comment]
Since my neologism was missed on gitter:

> I see there is scala/bug#11534 with a WIP. Probably the regression merged forward. What is the word for that? Pregression?

**Comment by Jasper-M** [comment]
Looks like this didn't get merged forward in time for 2.13.2.

**Comment by SethTisue** [comment]
hmm, that's odd. but in any case, HEAD of 2.13.x has the change.
